File: scripts/generate_icons.py
#!/usr/bin/env python3
"""
Скрипт для генерации иконок приложения Flow Max
Создает PNG иконки разных размеров для Android и iOS
"""

from PIL import Image, ImageDraw
import os

def create_icon(size, output_path):
    """Создает иконку заданного размера"""
    # Создаем новое изображение с прозрачным фоном
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Вычисляем масштаб
    scale = size / 1024
    offset = int(size * 0.2)
    
    # Рисуем фон с закругленными углами
    draw.rounded_rectangle([0, 0, size, size], radius=int(size * 0.2), 
                          fill=(52, 152, 219, 255))  # #3498DB
    
    # Рисуем котика (упрощенная версия)
    cat_center_x = int(312 * scale) + offset
    cat_center_y = int(280 * scale) + offset
    cat_radius = int(120 * scale)
    
    # Тело котика
    body_width = int(180 * scale)
    body_height = int(200 * scale)
    draw.ellipse([cat_center_x - body_width//2, cat_center_y + cat_radius - body_height//2,
                   cat_center_x + body_width//2, cat_center_y + cat_radius + body_height//2],
                  fill=(74, 144, 226, 255))  # #4A90E2
    
    # Голова
    draw.ellipse([cat_center_x - cat_radius, cat_center_y - cat_radius,
                   cat_center_x + cat_radius, cat_center_y + cat_radius],
                  fill=(74, 144, 226, 255))  # #4A90E2
    
    # Глаза
    eye_radius = int(25 * scale)
    left_eye_x = cat_center_x - int(32 * scale)
    right_eye_x = cat_center_x + int(32 * scale)
    eye_y = cat_center_y - int(20 * scale)
    
    # Белки глаз
    draw.ellipse([left_eye_x - eye_radius, eye_y - eye_radius,
                   left_eye_x + eye_radius, eye_y + eye_radius],
                  fill=(255, 255, 255, 255))
    draw.ellipse([right_eye_x - eye_radius, eye_y - eye_radius,
                   right_eye_x + eye_radius, eye_y + eye_radius],
                  fill=(255, 255, 255, 255))
    
    # Зрачки
    pupil_radius = int(15 * scale)
    draw.ellipse([left_eye_x - pupil_radius, eye_y - pupil_radius,
                   left_eye_x + pupil_radius, eye_y + pupil_radius],
                  fill=(44, 62, 80, 255))  # #2C3E50
    draw.ellipse([right_eye_x - pupil_radius, eye_y - pupil_radius,
                   right_eye_x + pupil_radius, eye_y + pupil_radius],
                  fill=(44, 62, 80, 255))  # #2C3E50
    
    # Нос
    nose_size = int(12 * scale)
    nose_points = [
        (cat_center_x, cat_center_y + int(10 * scale)),
        (cat_center_x - nose_size, cat_center_y + int(20 * scale)),
        (cat_center_x + nose_size, cat_center_y + int(20 * scale))
    ]
    draw.polygon(nose_points, fill=(231, 76, 60, 255))  # #E74C3C
    
    # Сохраняем изображение
    os.makedirs(os.path.dirname(output_path), exist_ok=True)
    img.save(output_path, 'PNG')
    print(f"Создана иконка: {output_path}")

def main():
    """Основная функция"""
    print("Генерация иконок для Flow Max...")
    
    # Размеры для Android
    android_sizes = {
        'mipmap-mdpi': 48,
        'mipmap-hdpi': 72,
        'mipmap-xhdpi': 96,
        'mipmap-xxhdpi': 144,
        'mipmap-xxxhdpi': 192
    }
    
    # Создаем иконки для Android
    for folder, size in android_sizes.items():
        output_path = f"android/app/src/main/res/{folder}/ic_launcher.png"
        create_icon(size, output_path)
    
    # Создаем иконку для iOS (1024x1024)
    ios_path = "ios/Flow_Max/Images.xcassets/AppIcon.appiconset/Icon-1024.png"
    create_icon(1024, ios_path)
    
    print("Все иконки созданы успешно!")
    print("\nДля использования:")
    print("1. Скопируйте сгенерированные PNG файлы в соответствующие папки")
    print("2. Для Android: android/app/src/main/res/mipmap-*/")
    print("3. Для iOS: ios/Flow_Max/Images.xcassets/AppIcon.appiconset/")

if __name__ == "__main__":
    main()

File: scripts/generate_icons.js
#!/usr/bin/env node
/**
 * Скрипт для генерации иконок приложения Flow Max
 * Создает простые PNG иконки разных размеров для Android и iOS
 */

const fs = require('fs');
const path = require('path');

// Простая функция для создания базовой иконки
function createSimpleIcon(size) {
    // Создаем простую иконку как base64 строку
    // Это будет простой синий квадрат с закругленными углами
    const canvas = `
<svg width="${size}" height="${size}" viewBox="0 0 ${size} ${size}" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <linearGradient id="grad" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#3498DB;stop-opacity:1" />
      <stop offset="50%" style="stop-color:#2980B9;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#1F4E79;stop-opacity:1" />
    </linearGradient>
  </defs>
  
  <!-- Фон -->
  <rect width="${size}" height="${size}" rx="${size * 0.2}" fill="url(#grad)"/>
  
  <!-- Котик (упрощенный) -->
  <g transform="translate(${size * 0.2}, ${size * 0.2}) scale(${size * 0.6 / 1024})">
    <!-- Тело -->
    <ellipse cx="312" cy="400" rx="180" ry="200" fill="#4A90E2"/>
    
    <!-- Голова -->
    <circle cx="312" cy="280" r="120" fill="#4A90E2"/>
    
    <!-- Глаза -->
    <circle cx="280" cy="260" r="25" fill="white"/>
    <circle cx="344" cy="260" r="25" fill="white"/>
    <circle cx="280" cy="260" r="15" fill="#2C3E50"/>
    <circle cx="344" cy="260" r="15" fill="#2C3E50"/>
    
    <!-- Нос -->
    <path d="M 312 290 L 300 300 L 324 300 Z" fill="#E74C3C"/>
    
    <!-- Лапки -->
    <ellipse cx="250" cy="580" rx="30" ry="40" fill="#357ABD"/>
    <ellipse cx="374" cy="580" rx="30" ry="40" fill="#357ABD"/>
  </g>
</svg>`;

    return canvas;
}

// Функция для создания папок
function ensureDirectoryExists(filePath) {
    const dirname = path.dirname(filePath);
    if (!fs.existsSync(dirname)) {
        fs.mkdirSync(dirname, { recursive: true });
    }
}

// Основная функция
function main() {
    console.log('Генерация иконок для Flow Max...');
    
    // Размеры для Android
    const androidSizes = {
        'mipmap-mdpi': 48,
        'mipmap-hdpi': 72,
        'mipmap-xhdpi': 96,
        'mipmap-xxhdpi': 144,
        'mipmap-xxxhdpi': 192
    };
    
    // Создаем иконки для Android
    for (const [folder, size] of Object.entries(androidSizes)) {
        const outputPath = path.join('android', 'app', 'src', 'main', 'res', folder, 'ic_launcher.svg');
        ensureDirectoryExists(outputPath);
        
        const iconSvg = createSimpleIcon(size);
        fs.writeFileSync(outputPath, iconSvg);
        console.log(`Создана иконка: ${outputPath}`);
    }
    
    // Создаем иконку для iOS
    const iosPath = path.join('ios', 'Flow_Max', 'Images.xcassets', 'AppIcon.appiconset', 'Icon-1024.svg');
    ensureDirectoryExists(iosPath);
    
    const iosIconSvg = createSimpleIcon(1024);
    fs.writeFileSync(iosPath, iosIconSvg);
    console.log(`Создана иконка: ${iosPath}`);
    
    console.log('\nВсе иконки созданы успешно!');
    console.log('\nДля использования:');
    console.log('1. SVG иконки созданы в соответствующих папках');
    console.log('2. Для Android: android/app/src/main/res/mipmap-*/');
    console.log('3. Для iOS: ios/Flow_Max/Images.xcassets/AppIcon.appiconset/');
    console.log('\nПримечание: Это SVG иконки. Для продакшена нужно конвертировать в PNG.');
}

// Запускаем скрипт
if (require.main === module) {
    main();
}

module.exports = { createSimpleIcon, ensureDirectoryExists };
